documentation/intrinsics.md

```diff
@@ -0,0 +1,658 @@
+RRSPACING(REAL(k) X) -> REAL(k)
+SCALE(REAL(k) X, INTEGER(any) I) -> REAL(k)
+SET_EXPONENT(REAL(k) X, INTEGER(any) I) -> REAL(k)
+SPACING(REAL(k) X) -> REAL(k)
+```
+
+### Restricted specific aliases for elemental conversions &/or extrema with default intrinsic types
+
+```
+AMAX0(INTEGER ...) = REAL(MAX(...))
+AMAX1(REAL ...) = MAX(...)
+AMIN0(INTEGER...) = REAL(MIN(...))
+AMIN1(REAL ...) = MIN(...)
+DBLE(REAL A) = REAL(A, KIND=KIND(0.0D0))
+DMAX1(DOUBLE PRECISION ...) = MAX(...)
+DMIN1(DOUBLE PRECISION ...) = MIN(...)
+FLOAT(INTEGER I) = REAL(I)
+IDINT(DOUBLE PRECISION A) = INT(A)
+IFIX(REAL A) = INT(A)
+MAX0(INTEGER ...) = MAX(...)
+MAX1(REAL ...) = INT(MAX(...))
+MIN0(INTEGER ...) = MIN(...)
+MIN1(REAL ...) = INT(MIN(...))
+SNGL(DOUBLE PRECISION A) = REAL(A)
+```
+
+### Generic elemental bit manipulation intrinsic functions
+```
+BGE(INTEGER(n1) or BOZ I, INTEGER(n2) or BOZ J) -> default LOGICAL
+BGT(INTEGER(n1) or BOZ I, INTEGER(n2) or BOZ J) -> default LOGICAL
+BLE(INTEGER(n1) or BOZ I, INTEGER(n2) or BOZ J) -> default LOGICAL
+BLT(INTEGER(n1) or BOZ I, INTEGER(n2) or BOZ J) -> default LOGICAL
+BTEST(INTEGER(n1) I, INTEGER(n2) POS) -> default LOGICAL
+DSHIFTL(INTEGER(k), INTEGER(k) or BOZ J, INTEGER(any) SHIFT) -> INTEGER(k)
+DSHIFTL(BOZ I, INTEGER(k), INTEGER(any) SHIFT) -> INTEGER(k)
+DSHIFTR(INTEGER(k), INTEGER(k) or BOZ J, INTEGER(any) SHIFT) -> INTEGER(k)
+DSHIFTR(BOZ I, INTEGER(k), INTEGER(any) SHIFT) -> INTEGER(k)
+IAND(INTEGER(k) I, INTEGER(k) or BOZ J) -> INTEGER(k)
+IAND(BOZ I, INTEGER(k) J) -> INTEGER(k)
+IBCLR(INTEGER(k) I, INTEGER(any) POS) -> INTEGER(k)
+IBSET(INTEGER(k) I, INTEGER(any) POS) -> INTEGER(k)
+IBITS(INTEGER(k) I, INTEGER(n1) POS, INTEGER(n2) LEN) -> INTEGER(k)
+IEOR(INTEGER(k) I, INTEGER(k) or BOZ J) -> INTEGER(k)
+IEOR(BOZ I, INTEGER(k) J) -> INTEGER(k)
+IOR(INTEGER(k) I, INTEGER(k) or BOZ J) -> INTEGER(k)
+IOR(BOZ I, INTEGER(k) J) -> INTEGER(k)
+ISHFT(INTEGER(k) I, INTEGER(any) SHIFT) -> INTEGER(k)
+ISHFTC(INTEGER(k) I, INTEGER(n1) SHIFT [, INTEGER(n2) SIZE = BIT_SIZE(I) ]) -> INTEGER(k)
+LEADZ(INTEGER(any) I) -> default INTEGER
+MASKL(INTEGER(any) I [, KIND=KIND(0) ]) -> INTEGER(KIND)
+MASKR(INTEGER(any) I [, KIND=KIND(0) ]) -> INTEGER(KIND)
+MERGE_BITS(INTEGER(k) I, INTEGER(k) or BOZ J, INTEGER(k) or BOZ MASK) = IOR(IAND(I,MASK),IAND(J,NOT(MASK)))
+MERGE_BITS(BOZ I, INTEGER(k) J, INTEGER(k) or BOZ MASK) = IOR(IAND(I,MASK),IAND(J,NOT(MASK)))
+NOT(INTEGER(k) I) -> INTEGER(k)
+POPCNT(INTEGER(any) I) -> default INTEGER
+POPPAR(INTEGER(any) I) -> default INTEGER = IAND(POPCNT(I), Z'1')
+SHIFTA(INTEGER(k) I, INTEGER(any) SHIFT) -> INTEGER(k)
+SHIFTL(INTEGER(k) I, INTEGER(any) SHIFT) -> INTEGER(k)
+SHIFTR(INTEGER(k) I, INTEGER(any) SHIFT) -> INTEGER(k)
+TRAILZ(INTEGER(any) I) -> default INTEGER
+```
+
+Any typeless "BOZ" argument to `MERGE_BITS` is converted to `INTEGER(k)`.  Note that at most one
+of its first two arguments can be a typeless "BOZ" literal.
```

Comment:
I think this note would be better at the top of the section. I was puzzled by the `BOZ` arguments and they weren't mentioned in the General Rules.

Reply:
Yes.